src/sage/algebras/commutative_dga.py

```diff
@@ -2699,17 +2730,28 @@ def extendy(phi, degree):
             B = A.cdg_algebra(A.differential({}))
             # Solve case that fails with one generator return B,gens
             phi = B.hom(gens)
-            phi = extendy(phi, degnzero + 1)
+            phiext = extendy(phi, degnzero + 1)
+            if isinstance(phiext, tuple):
+                if partial_result:
+                    return phiext[0]
+                else:
+                    raise ValueError("could not cover all relations in max iterations in degree {}".format(degnzero + 1))
```

Comment:
A more personal idiosyncrasy but
```suggestion
                if not partial_result:
                    raise ValueError("could not cover all relations in max iterations in degree {}".format(degnzero + 1))
                return phiext[0]
```
I find to be slightly easier to understand (or just removing the `else:`) as it more clearly indicates that it does terminate.